src/uct/ib/ud/base/ud_iface.h

```diff
@@ -257,10 +331,34 @@ static UCS_F_ALWAYS_INLINE void uct_ud_leave(uct_ud_iface_t *iface)
     UCS_ASYNC_UNBLOCK(iface->super.super.worker->async);
 }
 
+static UCS_F_ALWAYS_INLINE union ibv_gid*
+uct_ud_grh_get_dgid(struct ibv_grh *grh)
+{
+    uint8_t ipmask = 0xf0;
+    uint8_t ipver  = ((*(uint8_t*)grh) & ipmask) >> 4;
+    size_t i;
+
+    if (ipver != 6) {
+        /* for IPv6 GRH contains valid dgid value but for IPv4 there are only
+         * last UCS_IPV4_ADDR_LEN initialized bytes, reset begin to make hash
+         * function working */
+        for (i = 0; i < (sizeof(union ibv_gid) - UCS_IPV4_ADDR_LEN - 2);) {
+            grh->dgid.raw[i++] = 0x00;
+        }
```

Comment:
maybe use memset()?

Reply:
This loop has constant limit so it should be unrolled and transparent for any compiler optimization. Will check performance for both cases but I guess results will depends on compiler. Which one to check?